snuba/datasets/groups.py

```diff
@@ -48,14 +48,23 @@ def __init__(self) -> None:
 
         join_structure = JoinClause(
             left_node=TableJoinNode(
-                groupedmessage_source.format_from(),
-                groupedmessage_source.get_columns(),
-                self.GROUPS_ALIAS,
+                table_name=groupedmessage_source.format_from(),
+                columns=groupedmessage_source.get_columns(),
+                mandatory_conditions=[
+                    # TODO: This will be replaced as soon as expressions won't be strings
+                    # thus we will be able to easily add an alias to a column in an
+                    # expression.
+                    (qualified_column('record_deleted', self.GROUPS_ALIAS), '=', 0)
```

Comment:
Soon accessing a column in an expression will look like something like :

```
cols = expression.get_columns()
for c in cols:
      c.add_alias(alias)
      # (speculation, I still have to think through the syntax, but this is to explain the concept)
```

Right now, being the expression either a string or a sequence, extracting columns and adding the alias is quite complex, thus  think it is not worth building all the expression editing methods for strings when the structure is going to change very soon.